src/main/java/io/vertx/core/AsyncResult.java

```diff
@@ -59,4 +61,44 @@
    */
   boolean failed();
 
+  /**
+   * Apply a {@code mapper} function on this async result.<p>
+   *
+   * The {@code mapper} is called with the completed value and this mapper returns a value. This value will complete the result returned by this method call.<p>
+   *
+   * If the {@code mapper} throws an exception, the returned future will be failed with this exception.<p>
+   *
+   * When this async result is failed, the failure will be propagated to the returned future and the {@code mapper}
+   * will not be called.
+   *
+   * @param mapper the mapper function
+   * @return the mapped async result
+   */
+  <U> AsyncResult<U> map(Function<T, U> mapper);
+
+  /**
+   * Map the result of this async result to a specific {@code value}.<p>
+   *
+   * When this async result succeeds, this {@code value} will succeeeds the async result returned by this method call.<p>
+   *
+   * When this future fails, the failure will be propagated to the returned async result.
+   *
+   * @param value the value that eventually completes the mapped async result
+   * @return the mapped async result
+   */
+  <V> AsyncResult<V> map(V value);
+
+  /**
+   * Return the value if this async result is succeeded, otherwise the {@code other} value.
+   *
+   * @param other the other value to return
+   * @return the result when succeeded, otherwise the {@code other} value
+   */
+  default T orElse(T other) {
```

Comment:
I fear this is going to be a source of user problems. If the result is not known yet (AR neither failed nor succeeded), it will immediately return `other`.
But when I look at the Javadoc, I am wondering if this is the behavior you really wanted?